micro-rdk/src/common/config_monitor.rs

```diff
@@ -116,13 +117,35 @@ where
                         reboot = true;
                     }
                 }
+            }
+
+            if let Ok(device_agent_config) = app_client.get_agent_config().await {
+                if let Ok(agent_config) = AgentConfig::try_from(device_agent_config.as_ref()) {
+                    log::debug!("agent config: {:?}", agent_config);
+
+                    let stored = self.storage.get_network_settings().unwrap_or_default();
 
-                if reboot {
-                    // TODO(RSDK-9464): flush logs to app.viam before restarting
-                    self.restart();
+                    if agent_config.network_settings != stored {
+                        log::info!("new network settings found in config");
+                        if let Err(e) = self
+                            .storage
+                            .store_network_settings(&agent_config.network_settings)
+                        {
+                            log::error!("failed to store network settings to nvs: {}", e);
+                        } else {
+                            log::info!("successfully stored networks to nvs");
+                            reboot = true;
+                        }
+                    }
                 }
             }
 
+            if reboot {
+                log::info!("rebooting from config monitor...");
+                // TODO(RSDK-9464): flush logs to app.viam before restarting
+                self.restart();
```

Comment:
I think the reason we reboot for a robot configuration change is because we don't implement reconfiguration, so starting over is the only way to get a working robot again. But I don't think the same is necessarily true for wifi settings. On the other hand, implementing it without a reboot seems sort of complicated, since the updated settings might invalidate a network that is currently in use.

Reply:
yeah I wasn't sure about this part either. My reasoning for this was what if we get a network that has priority over the current connection and it doesn't reboot? For example, if a device is currently using a metered/costly network and we update to prioritize another one which isn't. Unless the metered network goes down or the device resets, the device won't switch to the non-metered network.

Reply:
I think that's a reasonable take. If/when we implement network roaming we will need to revisit this anyway.